File: Python Snippets/Fibonacci.py
#Function that return fibonacci series from a given integer number, using recursion

def Fibonacci(n):
    assert n > 0 and int(n) == n, "The number must be a positive integer only!"
    if n in [0, 1]:
        return n
    else:
        return Fibonacci(n-1) + Fibonacci(n-2)
        
print(Fibonacci(7))

File: TypeScriptModules/fibonacci.ts
import launchConsole from './utils/utils.ts'

const fibonacci = (n: number): number => {
  if (n in [0, 1]) return n
  else return fibonacci(n - 1) + fibonacci(n - 2)
}

launchConsole(
  'This module returns the fibonacci number of a given positive integer.\nPlease enter a positive integer: ',
  'The fibonacci number is',
  fibonacci
)

export default fibonacci
